### DISCUSSION TITLE
Full mojo app with variable in stash gives error

### DISCUSSION THREAD

**Discussion by pdkakoba** [mojolicious/mojo#1818] [Q&A]
From the tutorial-Stash and templates: the stash example works fine.  However in the full mojo app where in MyApp.pm:
`$r->get('/foo')->to('example#Foo');`
and in the controller, Example.pm:
`sub foo ($c) {
  $c->stash(one => 23);
  $c->render(template => 'foo', two => 24);
}`
and foo.html.eps:
`The magic numbers are <%= $one %> and <%= $two %>.`
I get the error msg: Global symbol "$one" requires explicit package name (did you forget to declare "my $one"?) at template example/Foo.html.ep line 1. and the same for $two.
This is in the environment: strawberry-perl 32, Win10 and Mojolicious 9.19
Where have I gone wrong?

**Comment by jberger** [comment]
That's a tricky one but I have a guess. Try routing to action "foo" not "Foo".  I suspect it isn't making it to the action but then auto rendering the template based on the lower casing of the action name. You can check this by putting an explicit log message in your controller, which I suspect you won't see; or by renaming the template to something not equal to the action name and using that name in render, that would break the automatic rendering and you'd get a 404 or something (I'm not at my computer to test right now)

**Reply by pdkakoba** [reply]
Great thanks, Joel, for your prompt reply.  Your suggestion using an explicit log message assisted me in finding the error -- a misnamed template.  Moreover, there was case-sensitive issue with the named action as you also pointed out.
